modules/core/localizer.js

```diff
@@ -220,24 +239,50 @@ export function coreLocalizer() {
 
         if (result !== undefined) {
           if (replacements) {
-            for (let k in replacements) {
-              const token = `{${k}}`;
-              const regex = new RegExp(token, 'g');
-              result = result.replace(regex, replacements[k]);
+            if (typeof result === 'object' && Object.keys(result).length) {
+                // If plural forms are provided, dig one level deeper based on the
+                // first numeric token replacement provided.
+                const number = Object.values(replacements).find(function(value) {
+                  return typeof value === 'number';
+                });
+                if (number !== undefined) {
+                  const rule = pluralRule(number, locale);
+                  if (result[rule]) {
+                    result = result[rule];
+                  } else {
+                    // We're pretty sure this should be a plural but no string
+                    // could be found for the given rule. Just pick the first
+                    // string and hope it makes sense.
+                    result = Object.values(result)[0];
+                  }
+                }
             }
+            if (typeof result === 'string') {
+              for (let k in replacements) {
+                const token = `{${k}}`;
+                const regex = new RegExp(token, 'g');
+                result = result.replace(regex, replacements[k]);
```

Comment:
This might be a good place to apply some [number formatting](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat) both for grouping characters (really big changesets!) and non-European numerals (for languages like Arabic and Hindi).

Reply:
Agreed, this is why I left strings like `{n} result for {search}` instead of `1 result for {search}`. I think I'll add formatting separately though.